youtube_dl/extractor/bbc.py

```diff
@@ -1074,6 +1086,85 @@ def _real_extract(self, url):
                     'subtitles': subtitles,
                 }
 
+        # simorgh-based playlist (see https://github.com/bbc/simorgh)
+        # JSON assigned to window.SIMORGH_DATA in a <script> element
+        simorgh_data = self._parse_json(
+            self._search_regex(
+                r'window\.SIMORGH_DATA\s*=\s*(\{[^<]+})\s*</',
+                webpage, 'simorgh playlist', default='{}'),
+            playlist_id, fatal=False)
+        # legacy media, video in promo object (eg, http://www.bbc.com/mundo/video_fotos/2015/06/150619_video_honduras_militares_hospitales_corrupcion_aw)
+        playlist = try_get(simorgh_data, lambda x: x['pageData']['promo']['media']['playlist']) or []
+        if playlist:
+            media = simorgh_data['pageData']['promo']
+            if media['media'].get('format') == 'video':
+                media.update(media['media'])
+                formats = []
+                keys = {'url', 'format', 'format_id', 'language', 'quality', 'tbr', 'resolution'}
+                for format in playlist:
+                    if not (format.get('url') and format.get('format')):
+                        continue
+                    bitrate = format.pop('bitrate')
+                    if bitrate:
+                        bitrate = re.sub(r'000\s*$', 'kbps', bitrate)
+                        format['tbr'] = parse_bitrate(bitrate)
+                    format['language'] = media.get('language')
+                    # format id: penultimate item from the url split on _ and .
+                    (fmt,) = re.split('[_.]', format['url'])[-2:][:1]
```

Comment:
Why not just use regular expression? It's much easier to understand and figure out the original intention after URL pattern changes in future.

Reply:
I'll look at that (time having passed), but what I wrote was my best attempt to be resilient against the format not having some resolution text (lo, med, hi) in that position.

Reply:
I tried various `re.match()` constructions and couldn't reproduce what the comment says is being done by the slice + re syntax. I'd rather leave this as it is.